t/t-repo-format.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+. "$(dirname "$0")/testlib.sh"
+
+begin_test "repository format version"
+(
+  set -e
+
+  reponame="lfs-repo-version"
+  git init $reponame
+  cd $reponame
+
+  [ -z "$(git config --local lfs.repositoryFormatVersion)" ]
+
+  git lfs track '*.dat'
+
+  [ "$(git config --local lfs.repositoryFormatVersion)" = "0" ]
+
+  git config --local lfs.repositoryFormatVersion 1
+
+  git lfs track '*.bin' >output 2>&1 && exit 1
+  cat output
```

Comment:
I'm guessing the `cat` is here in case of test failure, so it's a bit easier to debug what went wrong -- similar to the one in `t/t-locks.sh` discussed in https://github.com/git-lfs/git-lfs/pull/4446#discussion_r668771782.

Reply:
Yes, that's exactly the case.